src/VirtoCommerce.ContentModule.Data/Search/ContentSearchRequestBuilder.cs

```diff
@@ -64,14 +64,18 @@ protected virtual IList<IFilter> GetFilters(ContentSearchCriteria criteria)
                 result.Add(CreateTermFilter("StoreId", criteria.StoreId));
             }
 
-            if (!string.IsNullOrEmpty(criteria.LanguageCode))
-            {
-                result.Add(CreateTermFilter("CultureName", criteria.LanguageCode));
-            }
-            if (!string.IsNullOrEmpty(criteria.CultureName))
+            var cultureName = criteria.CultureName.IsNullOrEmpty()
+                ? criteria.LanguageCode
+                : criteria.CultureName;
```

Comment:
invert for better readability: if CultureName is not null then CultureName otherwise LanguageCode

Reply:
I've changed this logic, so the both values are used in the search filter